addons/mail/static/src/messaging/entity/messaging/messaging.js

```diff
@@ -53,24 +53,24 @@ function MessagingFactory({ Entity }) {
         async redirect({ id, model }) {
             if (model === 'mail.channel') {
                 const channel = this.env.entities.Thread.channelFromId(id);
-                if (!channel) {
+                if (!channel || !channel.isPinned) {
                     this.env.entities.Thread.joinChannel(id, { autoselect: true });
                     return;
                 }
                 channel.open();
             } else if (model === 'res.partner') {
-                if (id === this.currentPartner) {
+                if (id === this.currentPartner.id) {
                     this._openDocument({
                         model: 'res.partner',
                         id,
                     });
                     return;
                 }
                 const partner = this.env.entities.Partner.insert({ id });
-                if (partner.userId === undefined) {
+                if (!partner.user) {
```

Comment:
Shouldn't we add a test covering that?

Reply:
Is it in the scope of this PR you think ?

Reply:
If the fix is, the test is. But I'm all for making smaller commits, both can be done separately and that would be better.

Reply:
This is not resolved but I don't care enough about this to delay the PR.

@alexkuhn please have final word if we want/need a test or not, and if so, create task or say to do it here.

Reply:
No test in this PR, will make a subtask for that.

Reply:
task-2242786